presto-orc/src/test/java/com/facebook/presto/orc/TestOrcBloomfilters.java

```diff
@@ -0,0 +1,332 @@
+        assertEquals(bloomfilterWrite.getBitSize(), bloomfilters.get(0).getBloomfilter().getBitSize());
+        assertEquals(bloomfilterWrite.getNumHashFunctions(), bloomfilters.get(0).getBloomfilter().getNumHashFunctions());
+
+        // Validate bitset
+        int i = 0;
+        for (long l : bloomfilters.get(0).getBloomfilter().getBitSet()) {
+            assertEquals(l, bloomfilterWrite.getBitSet()[i]);
+            i++;
+        }
+
+        // Read directly: allows better inspection of the bitsets (helped to fix a lot of bugs)
+        CodedInputStream input = CodedInputStream.newInstance(bytes);
+        OrcProto.BloomFilterIndex deserializedBloomfilterIndex = OrcProto.BloomFilterIndex.parseFrom(input);
+        List<OrcProto.BloomFilter> bloomfilterList = deserializedBloomfilterIndex.getBloomFilterList();
+        assertEquals(1, bloomfilterList.size());
+
+        OrcProto.BloomFilter bloomFilterRead = bloomfilterList.get(0);
+
+        // Validate contents of ORC bloom filter bitset
+        i = 0;
+        for (long l : bloomFilterRead.getBitsetList()) {
+            assertEquals(l, bloomfilterWrite.getBitSet()[i]);
+            i++;
+        }
+        RowGroupBloomfilter rowGroupBloomfilter = new RowGroupBloomfilter(bloomFilterRead);
+
+        // Validate contents of bloom filter bitset
+        i = 0;
+        for (long l : rowGroupBloomfilter.getBloomfilter().getBitSet()) {
+            assertEquals(l, bloomfilterWrite.getBitSet()[i]);
+            i++;
+        }
+
+        BloomFilter rowGroupBloomfilterBloomfilter = rowGroupBloomfilter.getBloomfilter();
+        assertEquals(bloomfilterWrite.toString(), rowGroupBloomfilterBloomfilter.toString());
+
+        // hash functions
+        assertEquals(bloomfilterWrite.getNumHashFunctions(), rowGroupBloomfilterBloomfilter.getNumHashFunctions());
+        assertEquals(bloomfilterWrite.getNumHashFunctions(), bloomFilterRead.getNumHashFunctions());
+
+        // bit size
+        assertEquals(bloomfilterWrite.getBitSize(), rowGroupBloomfilterBloomfilter.getBitSize());
+        assertEquals(bloomfilterWrite.getBitSet().length, bloomFilterRead.getBitsetCount());
+
+        // test contents
+        assertFalse(rowGroupBloomfilterBloomfilter.testString(TEST_STRING_2));
+        rowGroupBloomfilterBloomfilter.addString(TEST_STRING_2);
+        assertTrue(rowGroupBloomfilterBloomfilter.testString(TEST_STRING_2));
+
+        // test contents
+        assertTrue(rowGroupBloomfilterBloomfilter.testString(TEST_STRING));
+        assertFalse(rowGroupBloomfilterBloomfilter.testString(TEST_STRING_NOT_WRITTEN));
+    }
+
+    @Test
+    public void testBloomfilterPredicateValues()
+            throws Exception
+    {
+        // set of tests, one pair per supported type, the "true" pairs will be written and the "false" pairs not
+        ImmutableMap.Builder<Object, Boolean> testBuilder = ImmutableMap.builder();
```

Comment:
Why not create two sets:
`entries` and `nonExistingEntries` (or call them as you wish).
Feed bloom filter with `entries` but test against both:
```java
for (entry : entries) {
// expect filter to find
}
for (entry : nonExistingEntries) {
// expect filter to not find
}
```

It would simplify the test vastly, removing this logic of bool flag in bunch of places.